### PR TITLE
feat: __options policy implementation

### LINKED ISSUE #61: add a specific key for policy options
at the moment, the definition for policy options is

```js
  policy: {
    Query: {
      welcome: {
        ttl: 5
      }
    }
  }
```

this could lead to potential conflicts if there are nested entities with the same name of policy options (ttl, skip, storage and so on), for example

```js
  policy: {
    Query: {
      welcome: {
        ttl: {
           ttl: 1,
           skip: () { /* .. */ }
        }
      }
    }
  }
```

the proposed solution would be to support the current configuration and introducing a specific key, for example '__options', to be used in case of conflicts, as follow

```js
  policy: {
    Query: {
      welcome: {
        // no __options key present, so policy options are considered as it is
        ttl: 6
      },
      hello: {
        // since "hello" query has a ttl property
        __options: {
          ttl: 6
        },
        ttl: {
          // here we can use both __options or list policy options
          skip: () { /* .. */ }
        }
      }
    }
  }
```

also, adding validation for nested policies here https://github.com/mercurius-js/cache/blob/main/lib/validation.js#L82

cc @codeflyer

### PR DESCRIPTION
**Pull request by tiagoheliob**
Implementing _options for nested fields conflicts

closes #61

### CODE REVIEW

**Review by mcollina — COMMENTED** [review]
Good work, code looks good, what's missing? (The PR is in draft)

**Comment by tiagoheliob** [comment]
@mcollina I was wondering if it needs extra unit tests, but since it looks good for you, I'll mark it as ready for review.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
@tiagoheliob I'm sorry but could you rebase? We have got a conflict now.

**Comment by tiagoheliob** [comment]
@mcollina done

**Review by simone-sanfratello — APPROVED** [review]

File: lib/validation.js [hunk 1]
```diff
@@ -123,6 +122,15 @@ function validateOpts (app, opts = {}) {
   return { all, policy, ttl, skip, storage, onDedupe, onHit, onMiss, onSkip, onError, logInterval, logReport }
 }
 
+// TODO need to validate also nested
```
[the remark this answers is not in the document]
Comment (tiagoheliob) [reply lib/validation.js:127]: @simone-sanfratello could you give me an example of a query that would need nested validation?

**Comment by simone-sanfratello** [comment]
let's merge and I'll open a new issue for nested policies

**Comment by mcollina** [comment]
Can you please fix the conflicts?

**Comment by tiagoheliob** [comment]
Sorry for the delay, it's done :heavy_check_mark:

**Pull request merged**